Assets/MRTK/SDK/Features/UX/Scripts/Buttons/ButtonConfigHelper.cs

```diff
@@ -414,7 +414,192 @@ public void ForceRefresh()
+                    if (!AssetDatabase.IsValidFolder(generatedIconSetFolder))
+                    {   // Create the folder if it doesn't exist
+                        AssetDatabase.CreateFolder(MixedRealityToolkitFiles.GetGeneratedFolder, customIconSetsFolderName);
+                    }
+
+                    string generatedIconSetPath = System.IO.Path.Combine(generatedIconSetFolder, generatedIconSetName + ".asset");
+                    targetIconSet = (ButtonIconSet)AssetDatabase.LoadAssetAtPath(generatedIconSetPath, typeof(ButtonIconSet));
+
+                    if (targetIconSet == null)
+                    {   // If the icon set doesn't already exist, duplicate the default
+                        ScriptableObject duplicateIconSet = Instantiate<ButtonIconSet>(defaultIconSet);
+                        duplicateIconSet.name = generatedIconSetName;
+
+                        AssetDatabase.CreateAsset(duplicateIconSet, generatedIconSetPath);
+                        AssetDatabase.SaveAssets();
+                        AssetDatabase.Refresh(ImportAssetOptions.ForceSynchronousImport);
+
+                        targetIconSet = (ButtonIconSet)AssetDatabase.LoadAssetAtPath(generatedIconSetPath, typeof(ButtonIconSet));
+                        createdIconSet = true;
+                    }
+                }
+            }
+
+            bool selectIconSet = false;
+            if (createdIconSet)
+            {
+                selectIconSet = EditorUtility.DisplayDialog("Custom Icon Set Created", string.Format(customIconSetCreatedMessage, generatedIconSetFolder), "View Asset", "OK");
+            }
+
+            try
+            {
+                // Set the icon set to the custom generated icon set
+                iconSetProp.objectReferenceValue = targetIconSet;
+                // Add the custom icon to the custom set
+                targetIconSet.EditorAddCustomQuadIcon(targetQuadIcon);
+                // Reset changes to the quad renderer
+                iconQuadTextureProp.objectReferenceValue = targetQuadIcon;
+                configObject.ApplyModifiedProperties();
+
+                // If the custom material shader is different from the default material, don't alter the material
+                if (targetQuadMaterial.shader.name == defaultButtonQuadMaterial.shader.name)
+                {   // If the custom material shader is the same, revert any prefab overrides
+                    SerializedObject iconQuadRendererObject = new SerializedObject(iconQuadRenderer);
+                    SerializedProperty materialsProp = iconQuadRendererObject.FindProperty("m_Materials");
+                    PrefabUtility.RevertPropertyOverride(materialsProp, InteractionMode.AutomatedAction);
+                }
+            }
+            catch (System.Exception e)
+            {
+                Debug.LogError("There was an error when converting your custom button icon.");
```

Comment:
If this hits, does this mean we'll have some iconset created that is now basically divorced from other things (i.e. it's not actually fully hooked up?). 

Usually for automated processes that hit issues, you want to add some form of rollback to ensure that we don't persist extra disconnected state (i.e. in this case assets we've created which are now not fully connected).

What's the particular exceptions that you're expecting to see here? i.e. what's the next step a user would take beyond seeing this message? Would it just be to file a github issue?

Reply:
I'm just going to remove the exception handling. It was based on a gut feeling, nothing concrete. And to date I haven't seen any exceptions halt the upgrade process while testing, in this section or others.